lib/src/firestore.dart

```diff
@@ -22,6 +22,14 @@ js.FieldPath createFieldPath(List<String> fieldNames) {
   return callConstructor(js.admin.firestore.FieldPath, jsify(fieldNames));
 }
 
+js.FieldValue fieldValueDelete() {
+  return js.admin.firestore.FieldValue.delete();
+}
+
+js.FieldValue fieldValueServerTimestamp() {
+  return js.admin.firestore.FieldValue.serverTimestamp();
+}
+
```

Comment:
These globally available functions bother me quite a bit. I see that you probably followed the pattern I started with `createGeoPoint` and others, so this is my bad.

I thought it would be nice to namespace them under Firestore and realized that we should be able to expose native `FieldValue` field like this:

```dart
/// Represents a Firestore Database and is the entry point for all
/// Firestore operations.
class Firestore {
  /// Sentinel field values that can be used when writing document fields with
  /// `set` or `update`.
  js.FieldValues get fieldValues => js.admin.firestore.FieldValue;

  // the rest of the Firestore class
}
```

So enum and global functions can go away. The `setFieldValue` method can declare it expects native `js.FieldValue` type and there is no need to convert between native values and enum.

WDYT?
If we do it this way I'll probably move other global functions under `Firestore` for consistency (after this PR is merged).

Reply:
you meant fieldValues to be static right (I assume so)? i.e. do you want to access them like this:
```
Firestore.fieldValues.serverTimestamp()
```
or
```
app.firestore().fieldValues.serverTimestamp()
```

> The setFieldValue method can declare it expects native js.FieldValue type

I thought you wanted to avoid exposing javascript types. Ok for me